In [60]:
import numpy as np

In [62]:
a = np.array([1, 2, 3, 4, 5, 6])
a

array([1, 2, 3, 4, 5, 6])

In [64]:
a = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
a

array([[ 1,  2,  3,  4],
       [ 5,  6,  7,  8],
       [ 9, 10, 11, 12]])

In [65]:
print(a[0])

[1 2 3 4]


In [66]:
np.zeros(2)

array([0., 0.])

In [67]:
np.ones(2)

array([1., 1.])

In [70]:
np.arange(4)

array([0, 1, 2, 3])

In [71]:
np.arange(2, 9, 2)

array([2, 4, 6, 8])

In [72]:
x = np.ones(2, dtype=np.int64)

In [73]:
x

array([1, 1])

In [74]:
arr = np.array([2, 1, 5, 3, 7, 4, 6, 8])

In [75]:
np.sort(arr)

array([1, 2, 3, 4, 5, 6, 7, 8])

In [76]:
a = np.array([1, 2, 3, 4])
b = np.array([5, 6, 7, 8])


In [77]:
np.concatenate((a, b))

array([1, 2, 3, 4, 5, 6, 7, 8])

In [78]:
x = np.array([[1, 2], [3, 4]])
y = np.array([[5, 6]])


In [81]:
np.concatenate((x, y), axis=0)

array([[1, 2],
       [3, 4],
       [5, 6]])

In [82]:
x.ndim

2

In [83]:
x.size

4

In [84]:
x.shape

(2, 2)

In [85]:
a = np.arange(6)

In [86]:
print(a)

[0 1 2 3 4 5]


In [87]:
b = a.reshape(3, 2)
print(b)

[[0 1]
 [2 3]
 [4 5]]


In [88]:
data = np.array([1, 2, 3])
data[1]



2

In [89]:
data[0:2]


array([1, 2])

In [90]:
data[1:]


array([2, 3])

In [91]:
data[-2:]

array([2, 3])

In [92]:
data = np.array([1, 2])
ones = np.ones(2, dtype=int)
data + ones


array([2, 3])

In [93]:
data - ones

array([0, 1])

In [94]:
data * data

array([1, 4])

In [95]:
data / data


array([1., 1.])

In [96]:
a = np.array([1, 2, 3, 4])

In [97]:
a.sum()

10

In [98]:
b = np.array([[1, 1], [2, 2]])

In [99]:
b

array([[1, 1],
       [2, 2]])

In [100]:
b.sum(axis=0)

array([3, 3])

In [101]:
b.sum(axis=1)

array([2, 4])

In [102]:
b*5

array([[ 5,  5],
       [10, 10]])

In [103]:
data.max()

2

In [104]:
data.min()

1

In [105]:
data.sum()

3

In [106]:
data = np.array([[1, 2], [3, 4], [5, 6]])
data

array([[1, 2],
       [3, 4],
       [5, 6]])

In [107]:
data[0, 1]

2

In [108]:
data[1:3]

array([[3, 4],
       [5, 6]])

In [109]:
a = np.array([11, 11, 12, 13, 14, 15, 16, 17, 12, 13, 11, 14, 18, 19, 20])

In [110]:
unique_values = np.unique(a)

In [111]:
print(unique_values)

[11 12 13 14 15 16 17 18 19 20]


In [112]:
arr = np.array([1, 2, 3, 4, 5, 6, 7, 8])

In [113]:
reversed_arr = np.flip(arr)

In [114]:
print('Reversed Array: ', reversed_arr)

Reversed Array:  [8 7 6 5 4 3 2 1]


In [115]:
# Importing pandas library
import pandas as pd

# Creating and initializing a nested list
age = [['Aman', 95.5, "Male"], ['Sunny', 65.7, "Female"],
       ['Monty', 85.1, "Male"], ['toni', 75.4, "Male"]]

# Creating a pandas dataframe
df = pd.DataFrame(age, columns=['Name', 'Marks', 'Gender'])

# Printing dataframe
df


Name  Marks  Gender
0   Aman   95.5    Male
1  Sunny   65.7  Female
2  Monty   85.1    Male
3   toni   75.4    Male

In [116]:
# Importing Numpy package
import numpy as np

# Creating a 3-D numpy array using np.array()
org_array = np.array([[23, 46, 85],[43, 56, 99],[11, 34, 55]])

# Printing the Numpy array
print(org_array)


[[23 46 85]
 [43 56 99]
 [11 34 55]]





## Pandas Dataframe  vs NumPy arrays/matrices


|Pandas Dataframe               |NumPy arrays/matrices                |
|-------------------------------|-----------------------------|
|When we have to work on **Tabular data**, we prefer the pandas module.            |When we have to work on **Numerical data**, we prefer the numpy module.            |
|The powerful tools of pandas are **Data frame and Series.**           |Whereas the powerful tool of _numpy_ is **Arrays.**           |
|Pandas consume **more**  **memory**.|_Numpy_ is **memory efficient.**|
|Pandas has a better performance when a number of rows is  **500K or more.** |Numpy has a better performance when number of rows is  **50K or less.**|
|Indexing of the  _pandas_  series is  **very slow**  as compared to  _numpy_  arrays.|Indexing of  numpy  Arrays is **very fast**.|
|Pandas  offer a have2d table object called  **DataFrame.**|Numpy is capable of providing  **multi-dimensional arrays.**|



In [117]:
data = [1, 4, 5, 7, 8, 8, 10]
#calculate average value of array
np.mean(data)


6.142857142857143

In [118]:

#calcualte average value of array
np.average(data)

6.142857142857143

In [119]:
np.average(data, weights=(.1, .2, .4, .05, .05, .1, .1))

5.449999999999999

Here is the formula that np.average() used to calculate this value:

Weighted Average = 1* .1 + 4* .2 + 5* .4 + 7* .05 + 8* .05 + 8* .1 + 10* .1 = 5.45.


### np.average takes an optional weight parameter. If it is not supplied it is same as np.mean